packages/dev/core/src/Materials/Node/Blocks/Dual/currentScreenBlock.ts

```diff
@@ -189,17 +191,28 @@ export class CurrentScreenBlock extends NodeMaterialBlock {
             if (state.target === NodeMaterialBlockTargets.Fragment) {
                 return;
             }
+            const textureReadFunc =
+                state.shaderLanguage === ShaderLanguage.GLSL
+                    ? `texture2D(${this._samplerName},`
+                    : `textureSampleLevel(${this._samplerName}, ${this._samplerName + Constants.AUTOSAMPLERSUFFIX},`;

```

Comment:
Why not using `textureSample` instead of `textureSampleLevel`? The result will be different between WebGL and WebGPU because of this.

Is it because of uniformity analysis errors? If so, in which case does this happen (I think it should not happen in a node material, as we don't generate conditional branches)?

Reply:
Well because our lovely WebGPU does not support textureSample in a vertexShader:
![image](https://github.com/BabylonJS/Babylon.js/assets/1306056/d7a18dde-6b15-4d1a-8bf7-4034d790ccf6)

Reply:
https://webgpu.rocks/wgsl/functions/texture/

Reply:
Ah yes, of course (there's no derivatives in a vertex shader)!
Now I wonder why it works in WebGL (I guess Angle replaces texture2D with textureLOD)...